contract-hardhat/test/o_load_massive_investor_flow.ts

```diff
@@ -0,0 +1,965 @@
+
+            expect(tradingRestrictionEvent).to.not.be.null;
+            expect(tradingRestrictionEvent!.args.newManager).to.equal(I_TradingRestrictionManager.target, "TradingRestrictionManager not set correctly");
+        });
+        it("Should whitelist and issue tokens to many investors", async () => {
+            const ltime = await latestTime();
+            const expiry = ltime + duration.days(300);
+
+            smallInvestors = accounts.slice(1, 901); // 900 small
+            largeInvestors = accounts.slice(901, 1001); // 100 large
+            allInvestors = [...smallInvestors, ...largeInvestors];
+            console.log(`Whitelisting ${allInvestors.length} investors...`);
+
+            for (const investor of allInvestors) {
+                const isAccredited = Math.random() < 0.5; // random accreditation
+                const investorClass = getRandomInvestorClass();
+                investorClassMap[investor.address.toLowerCase()] = investorClass;
+
+                values.push([investor.address, BigInt(expiry), isAccredited]);
+            }
+
+
+            const merkleTree = StandardMerkleTree.of(values, ["address", "uint64", "bool"]);
+            const merkleRoot = merkleTree.root;
+
+            await I_TradingRestrictionManager.connect(token_owner).modifyKYCData(merkleRoot);
+
+            for (const [index, [address, expiry, isAccredited]] of merkleTree.entries()) {
+
+                const proof = merkleTree.getProof(index);
+                const signer = accounts.find(acc => acc.address === address)!;
+
+                const investorClass = investorClassMap[address.toLowerCase()];
+                // Whitelist investor
+                await expect(
+                    I_TradingRestrictionManager.connect(signer).verifyInvestor(
+                        proof,
+                        address,
+                        expiry,
+                        isAccredited,
+                        investorClass
+                    )
+                ).to.not.be.reverted;
+
+                //Issue tokens based on group
+                const rawAmount = smallInvestors.includes(signer)
+                    ? randomInt(10, 5000)         // small: 10–5,000
+                    : randomInt(10_000, 500_000);  // large holder
+
+                const amount = ethers.parseEther(rawAmount.toString());
+
+                await I_SecurityToken.connect(token_owner).issue(
+                    address,
+                    amount,
+                    "0x"
+                );
+
+                //validate balance
+                const balance = await I_SecurityToken.balanceOf(address);
+                expect(balance).to.equal(amount);
+
+                issuedAmounts[address.toLowerCase()] = amount;
+            }
+
+            console.log(`Verified & issued tokens to ${allInvestors.length} investors using Merkle Tree.`);
+        });
+
+        it("Should allow existing investors to buy more tokens", async () => {
```

Comment:
Investors should acquire token through buyWithUSD(STO) not through issue. This it reflect what we have in our system